src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,149 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        tail = new Node<>(null, null, null);
+        head = new Node<>(null, null, null);
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(null, value, null);
+        if (size == 0) {
+            tail = head = node;
+        } else {
+            tail.next = node;
+            node.prev = tail;
+            tail = node;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index);
```

Comment:
```suggestion
        checkIndex(index);
```
i think maybe remove this method at all and write checking condition here, because you use this method just one time in your code. i dont think if its necessary to have it